### PR TITLE
Tests/remove network dependent

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Removes and adjusts unit tests, so they are not network-dependent. Unit tests should not rely on remote services and servers.

### How to test the changes in this Pull Request:

1. Observe the tests are passing on CI and locally

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: tests/unit-tests/api-plugins-controller.php [hunk 1]
```diff
@@ -85,159 +82,6 @@ public function test_get_plugins_authorized() {
 		$this->assertEquals( $expected_jetpack_info, $data['jetpack'] );
 	}
 
-	/**
```
Comment (leogermani) [inline tests/unit-tests/api-plugins-controller.php:88]: afaik, all these tests are not making any requests. `server->dispatch` is emulating the server receiving a request

Comment (leogermani) [reply tests/unit-tests/api-plugins-controller.php:88]: It's the way to test Rest API endpoints without needing http requests

File: tests/unit-tests/emails.php [hunk 2]
```diff
@@ -51,26 +51,6 @@ private static function get_test_email( $type ) {
 	 * Email setup & defaults generation.
 	 */
 	public function test_emails_setup() {
-		self::assertEquals(
-			Emails::get_emails( [ 'test-email-config' ] ),
-			[],
-			'Emails are empty until configured.'
-		);
-		self::assertFalse(
-			Emails::can_send_email( 'test-email-config' ),
```
Comment (leogermani) [inline tests/unit-tests/emails.php:60]: can we short circuit `wp_mail` to make sure no requests are made?

File: tests/unit-tests/api-plugins-controller.php (hunk 1 above)
Comment (adekbadek) [reply tests/unit-tests/api-plugins-controller.php:88]: It's not about testing the REST API endpoints. The endpoint _handler_ is making requests to the plugin repository to fetch plugin code.

File: tests/unit-tests/emails.php (hunk 2 above)
Comment (adekbadek) [reply tests/unit-tests/emails.php:60]: This is not about `wp_mail`. The issue with this test is that it installed `newspack-newsletters` from the plugin repository. I've replaced it with a mock of `Newspack_Newsletters` class, and to simplify removed the test assertions related to the plugin being inactive. This is not ideal, I know, but I don't think making it ideal is worth the effort.

File: tests/unit-tests/api-plugins-controller.php (hunk 1 above)
Comment (leogermani) [reply tests/unit-tests/api-plugins-controller.php:88]: gotcha

**Review by leogermani — APPROVED** [review]

**Pull request merged**